compiler/rustc_trait_selection/src/error_reporting/traits/suggestions.rs

```diff
@@ -5108,6 +5108,38 @@ impl<'a, 'tcx> TypeErrCtxt<'a, 'tcx> {
             _ => {}
         }
     }
+    pub(crate) fn suggest_impl_similarly_named_trait(
+        &self,
+        err: &mut Diag<'_>,
+        trait_predicate: ty::PolyTraitPredicate<'tcx>,
+    ) {
+        let trait_def_id = trait_predicate.def_id();
+        let trait_name = self.tcx.item_name(trait_def_id);
+        if let Some(other_trait_def_id) = self.tcx.all_traits_including_private().find(|def_id| {
+            trait_def_id != *def_id
+                && trait_name == self.tcx.item_name(def_id)
+                && self
+                    .tcx
+                    .all_impls(*def_id)
+                    .find(|impl_def_id| {
+                        let imp = self.tcx.impl_trait_header(impl_def_id).unwrap();
+                        if imp.polarity != ty::ImplPolarity::Positive {
+                            return false;
+                        }
+                        let imp = imp.trait_ref.skip_binder();
+                        trait_predicate.skip_binder().self_ty() == imp.self_ty()
+                    })
+                    .is_some()
```

Comment:
this only catches impls where the self type matches exactly, use `predicate_must_hold_modulo_regions` instead

Reply:
Good point. Yeah, I was not completly sure. Thanks for your review, I gonna to take a look at this method.